rom c0754bdf383708b439fc7611346b13c83b86b800 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jani=20V=C3=A4limaa?= <wally@mageia.org>
Date: Mon, 13 Mar 2017 21:10:29 +0200
Subject: [PATCH] Fix gtk-doc build with libtool

LT_INIT should be introduced before GTK_DOC_CHECK in configure.ac to
make gtk-doc build to use libtool.

This also fixes the gtk-doc build with --disable-static.
---
 configure.ac               | 22 +++++++++++-----------
 docs/reference/Makefile.am |  2 +-
 2 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index db0ecff..1444b98 100644
--- a/configure.ac
+++ b/configure.ac
@@ -44,6 +44,17 @@ AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([-Wall -Werror -Wno-portability foreign silent-rules subdir-objects])
 AM_SILENT_RULES([yes])
 
+# Checks for programs.
+AC_PROG_CC
+AM_PROG_AR
+
+LT_INIT
+LIB_VERSION_INFO="lib_current:lib_revision:lib_age"
+AC_SUBST(LIB_VERSION_INFO)
+
+GIR_VERSION="gir_version"
+AC_SUBST(GIR_VERSION)
+
 # Option to make doc
 m4_ifdef([GTK_DOC_CHECK], [
 GTK_DOC_CHECK([1.19],[--flavour no-tmpl])
@@ -57,17 +68,6 @@ AC_PATH_PROG([XSLTPROC], [xsltproc])
 AM_CONDITIONAL([ENABLE_GTK_DOC], false)
 ])
 
-# Checks for programs.
-AC_PROG_CC
-AM_PROG_AR
-
-LT_INIT
-LIB_VERSION_INFO="lib_current:lib_revision:lib_age"
-AC_SUBST(LIB_VERSION_INFO)
-
-GIR_VERSION="gir_version"
-AC_SUBST(GIR_VERSION)
-
 # Option for example
 AC_ARG_ENABLE([example],
 	AS_HELP_STRING([--enable-example], [enable the example]),
diff --git a/docs/reference/Makefile.am b/docs/reference/Makefile.am
index 85b49d3..05e7303 100644
--- a/docs/reference/Makefile.am
+++ b/docs/reference/Makefile.am
@@ -70,7 +70,7 @@ expand_content_files=
 # e.g. GTKDOC_CFLAGS=-I$(top_srcdir) -I$(top_builddir) $(GTK_DEBUG_FLAGS)
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
 GTKDOC_CFLAGS=$(DEP_CFLAGS)
-GTKDOC_LIBS=$(DEP_LIBS) $(top_builddir)/.libs/libstatusnotifier.a
+GTKDOC_LIBS=$(top_builddir)/.libs/libstatusnotifier.la
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
 include $(top_srcdir)/gtk-doc.make
-- 
2.10.2

